README.md
# ubuntu-18.04-awx

Vagrant Box with Ubuntu 18.04 and awx

## Base image

Used base image [elegoev/ubuntu-18.04](https://app.vagrantup.com/elegoev/boxes/ubuntu-18.04)

## Directory Description

| directory | description                                          |
|-----------|------------------------------------------------------|
| inspec    | inspec test profiles with controls                   |
| packer    | packer build, provisioner and post-processor scripts |
| test      | test environment for provision & inspec development  |

## Vagrant

### Vagrant Cloud

- [elegoev/ubuntu-18.04-awx](https://app.vagrantup.com/elegoev/boxes/ubuntu-18.04-awx)

### Vagrant Plugins

- [vagrant-disksize](https://github.com/sprotheroe/vagrant-disksize)
- [vagrant-hosts](https://github.com/oscar-stack/vagrant-hosts)
- [vagrant-secret](https://github.com/tcnksm/vagrant-secret)
- [vagrant-vbguest](https://github.com/dotless-de/vagrant-vbguest)
- [vagrant-serverspec](https://github.com/vvchik/vagrant-serverspec)
- [vagrant-vmware-esxi](https://github.com/josenk/vagrant-vmware-esxi)

### Vagrantfile

    Vagrant.configure("2") do |config|

      $basebox_name="ubuntu-18.04-awx-test"
      $basebox_hostname="ubuntu-1804-awx-test"
      $src_image_name="elegoev/ubuntu-18.04-awx"
      $vb_group_name="basebox-awx-test"

      config.vm.define "#{$basebox_name}" do |machine|
        machine.vm.box = "#{$src_image_name}"
    
        # define guest hostname
        machine.vm.hostname = "#{$basebox_hostname}"

        machine.vm.provider "virtualbox" do |vb|
          vb.name = $basebox_name
          vb.cpus = 1
          vb.customize ["modifyvm", :id, "--memory", "1024" ]
          vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
          vb.customize ["modifyvm", :id, "--natdnsproxy1", "on"]
          vb.customize [
            "modifyvm", :id, "--uartmode1", "file",
            File.join(Dir.pwd, "ubuntu-bionic-18.04-cloudimg-console.log")
          ]
          vb.customize ["modifyvm", :id, "--groups", "/#{$vb_group_name}" ]
          vb.customize ["modifyvm", :id, "--vram", 256 ]
        end

      end   

    end

## References

- [How to Install Ansible AWX on Ubuntu 20.04 LTS](https://www.linuxtechi.com/install-ansible-awx-on-ubuntu/)

## Versioning

Repository follows sematic versioning  [![semantic versioning](https://img.shields.io/badge/semver-2.0.0-green.svg)](http://semver.org)

## Changelog

For all notable changes see [CHANGELOG](https://github.com/elegoev/basebox-ubuntu-18.04-awx/blob/main/CHANGELOG.md)

## License

Licensed under The MIT License (MIT) - for the full copyright and license information, please view the [LICENSE](https://github.com/elegoev/basebox-ubuntu-18.04-awx/blob/main/LICENSE) file.

## Issue Reporting

Any and all feedback is welcome.  Please let me know of any issues you may find in the bug tracker on github. You can find it [here.](https://github.com/elegoev/basebox-ubuntu-18.04-awx/issues)


CHANGELOG.md
# Change Log

## 1.0.x

**Implemented enhancements:**

- AWX installation


packer/provisioners/base-provisioning.sh
##!/bin/bash

application_file_path="/vagrant/installed-application.md"

# install ansible
#

# install ansible repository
sudo apt-add-repository ppa:ansible/ansible
sudo apt update
sudo apt install python-pip -y
sudo pip install "pywinrm>=0.2.2"
sudo apt install ansible -y
sudo pip install cryptography==2.2.2

# install nodejs
#
# sudo npm install npm --global

# install docker-compose
#
DOCKER_VERSION="1.29.0"
# sudo pip3 install docker-compose==1.28.5
#
sudo curl -L "https://github.com/docker/compose/releases/download/${DOCKER_VERSION}/docker-compose-$(uname -s)-$(uname -m)" \
          -o /usr/local/bin/docker-compose
sudo chmod +x /usr/local/bin/docker-compose
sudo ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose

# # install awx
# sudo apt install -y python3-pip git pwgen unzip
# wget https://github.com/ansible/awx/archive/17.1.0.zip
# unzip 17.1.0.zip
# cd ./awx-17.1.0/installer
# pwgen -N 1 -s 30
# # TifPnqD2ycLSS1fpIhoxDjb8eMD7sO
# # vi inventory
# # admin_user=admin
# # admin_password=<Strong-Admin-password>
# # secret_key=lKjpI3Hdj2PWlp8De6g2pDj9e5dU5e
# # ansible-playbook -i inventory install.yml


# set ANSIBLE_CONFIG
# see https://docs.ansible.com/ansible/devel/reference_appendices/config.html#cfg-in-world-writable-dir
sudo echo "export ANSIBLE_CONFIG='./ansible.cfg'" >> /home/vagrant/.bashrc

ANSIBLE_VERSION=$(ansible --version | sed -n 1p | grep ansible | awk  '{print $2}')
DOCKERCOMPOSE_VERSION=$(docker-compose --version | awk  '{print $3}')
echo "# Installed application   "  > $application_file_path
echo "***                       " >> $application_file_path
echo "> ansible ${ANSIBLE_VERSION}" >> $application_file_path
echo "> docker-compose ${DOCKERCOMPOSE_VERSION}" >> $application_file_path


packer/post-processors/create-git-tag.ps1
# create git tag

$src_image_version = $env:IMAGE_VERSION

$vagrant_file_template = '
Vagrant.configure("2") do |config|

  $basebox_name="elegoev-ubuntu-18.04-awx-test"
  $basebox_hostname="ubuntu-1804-awx-test"
  $src_image_name="elegoev/ubuntu-18.04-awx"
  $vb_group_name="basebox-awx-test"

  config.vm.define "#{$basebox_name}" do |machine|
    machine.vm.box = "#{$src_image_name}"
    machine.vm.box_version = "$image_version"
    
    # define guest hostname
    machine.vm.hostname = "#{$basebox_hostname}"

    machine.vm.provider "virtualbox" do |vb|
      vb.name = $basebox_name
      vb.cpus = 1
      vb.customize ["modifyvm", :id, "--memory", "1024" ]
      vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
      vb.customize ["modifyvm", :id, "--natdnsproxy1", "on"]
      vb.customize ["modifyvm", :id, "--uartmode1", "disconnected" ]
      vb.customize ["modifyvm", :id, "--groups", "/#{$vb_group_name}" ]
      vb.customize ["modifyvm", :id, "--vram", 256 ]
    end

    machine.trigger.after :destroy do |trigger|
      trigger.info = "Cleanup environment"
      trigger.run = {path: "./cleanup-vagrant.ps1"}  
    end

  end   

end
'

# create Vagrantfile for download test
$vagrant_file_expanded = $vagrant_file_template.Replace('$image_version', "$src_image_version")
Set-Content -Path './test/download/Vagrantfile' -Value "$vagrant_file_expanded"

# create git tag
git add --all
git commit -m "new basebox version $src_image_version"
git push 
git tag $src_image_version
git push --tags





